engine/api/src/main/java/io/deephaven/engine/table/vectors/ColumnVectors.java

```diff
@@ -0,0 +1,489 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.vectors;
+
+import io.deephaven.engine.rowset.TrackingRowSet;
+import io.deephaven.engine.table.ColumnDefinition;
+import io.deephaven.engine.table.ColumnSource;
+import io.deephaven.engine.table.Table;
+import io.deephaven.vector.ByteVector;
+import io.deephaven.vector.CharVector;
+import io.deephaven.vector.DoubleVector;
+import io.deephaven.vector.FloatVector;
+import io.deephaven.vector.IntVector;
+import io.deephaven.vector.LongVector;
+import io.deephaven.vector.ObjectVector;
+import io.deephaven.vector.ShortVector;
+import io.deephaven.vector.Vector;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Utility methods for constructing {@link Vector Vectors} from the columns of a {@link Table}, enabling inefficient
+ * random-access and efficient bulk access to column data by row position.
+ */
+public final class ColumnVectors {
+
+    private ColumnVectors() {}
+
+    /**
+     * Get a {@link Vector} of the data belonging to the specified column. This is useful for randomly-accessing column
+     * data by row position (which is fairly inefficient), or extracting data from the column in bulk (which is quite
+     * efficient).
+     * <p>
+     * If {@code table} is {@link Table#isRefreshing() refreshing}, the returned vector is only valid for use during the
```

Comment:
I think there should be an "easy" button for that.  Possibilities include ignoring the cycle for immutable columns; or making a snapshot copy. It might not matter because you run it and then look at it under the lock most often as well, but then this becomes scary from the API standpoint and you could suggest that.

Reply:
Also we have huge amounts of repetitive documentation.  It might make sense to have greater exposition at the class level, with terse methods that have links to the top.

Reply:
Neither column iterators nor DataColumns provide(d) an easy button, nor is an easy button necessary for use from listeners (the only safe place to use prev, anyway).
If users want to snapshot, they can.